engines/sci/graphics/palette32.cpp

```diff
@@ -21,20 +21,156 @@
  */
 
 #include "common/file.h"
+#include "common/system.h"
+#include "graphics/palette.h"
 
 #include "sci/sci.h"
+#include "sci/resource.h"
 #include "sci/graphics/palette32.h"
+#include "sci/graphics/screen.h"
 
 namespace Sci {
 	
 GfxPalette32::GfxPalette32(ResourceManager *resMan, GfxScreen *screen)
-	: GfxPalette(resMan, screen), _clutTable(0), _cyclers(), _cycleMap() {}
+	: GfxPalette(resMan, screen),
+	_clutTable(nullptr),
+	// Palette cycling
+	_cyclers(), _cycleMap(),
+	// Palette varying
+	_varyTime(0), _varyDirection(0), _varyTargetPercent(0),
+	_varyTargetPalette(nullptr), _varyStartPalette(nullptr),
+	_varyFromColor(0), _varyToColor(255), _varyNumTimesPaused(0),
+	_varyPercent(_varyTargetPercent), _varyLastTick(0),
+	// Palette versioning
+	_version(1), _needsUpdate(false) {
+		// TODO: Sci::Palette should have a default constructor that does
+		// all of this
+		for (int color = 0; color < ARRAYSIZE(_sysPalette.colors); color++) {
+			_sourcePalette.colors[color] = _nextPalette.colors[color] = _sysPalette.colors[color];
+			// TODO: Intensity and mapping are not used by SCI32 but are used by
+			// other parts of the graphic engine right now. Replace those parts.
```

Comment:
I don't understand this comment. Intensity is used by older SCI games, called by kPaletteSetIntensity. Mapping is used by older SCI games, which merged the palette of sprites with the palette of the actual scene. This functionality is not used by SCI32 at all, but it is used by the rest of the SCI engine, so I don't understand why the logic in those parts should be replaced...